contracts/PercentageTransferManager/src/lib.rs

```diff
@@ -0,0 +1,593 @@
+            {
+                return RestrictionResult::Invalid;
+            } else {
+                return RestrictionResult::Valid;
+            }
+        }
+
+        /// Change the value of allowed percentage
+        ///
+        /// # Arguments
+        /// * `new_percentage` - New value of Max percentage of assets hold by an investor
+        #[ink(message)]
+        fn change_allowed_percentage(&mut self, new_percentage: u128) {
+            assert!(self.env().caller() == *self.owner.get(), "Not Authorized");
+            assert!(
+                *self.max_allowed_percentage.get() != new_percentage,
+                "Must change setting"
+            );
+            self.env().emit_event(ChangeAllowedPercentage {
+                old_percentage: *self.max_allowed_percentage.get(),
+                new_percentage: new_percentage,
+            });
+            self.max_allowed_percentage.set(new_percentage);
+        }
+
+        /// Sets whether or not to consider primary issuance transfers
+        ///
+        /// # Arguments
+        /// * `primary_issuance` - whether to allow all primary issuance transfers
+        #[ink(message)]
+        fn change_primary_issuance(&mut self, primary_issuance: bool) {
+            assert!(self.env().caller() == *self.owner.get(), "Not Authorized");
+            assert!(
+                *self.allow_primary_issuance.get() != primary_issuance,
+                "Must change setting"
+            );
+            self.allow_primary_issuance.set(primary_issuance);
+            self.env().emit_event(ChangePrimaryIssuance {
+                allow_primary_issuance: primary_issuance,
+            });
+        }
+
+        /// To exempt the given Identity from the restriction
+        ///
+        /// # Arguments
+        /// * `identity` - Identity of the token holder whose exemption status needs to change
+        /// * `is_exempted` - New exemption status of the identity
+        #[ink(message)]
+        fn modify_exemption_list(&mut self, identity: IdentityId, is_exempted: bool) {
+            assert!(self.env().caller() == *self.owner.get(), "Not Authorized");
+            assert!(
+                self._is_exempted_or_not(&identity) != is_exempted,
+                "Must change setting"
+            );
+            self._modify_exemption_list(identity, is_exempted);
+        }
+
+        /// To exempt the given Identities from the restriction
+        ///
+        /// # Arguments
+        /// * `identities` - Identities of the token holders whose exemption status needs to change
+        /// * `exemptions` - New exemption status of the identities
+        #[ink(message)]
+        fn modify_exemption_list_batch(
+            &mut self,
+            identities: Vec<IdentityId>,
+            exemptions: Vec<bool>,
+        ) {
+            assert!(
+                identities.len() == exemptions.len(),
+                "Arguments length mismatch"
+            );
```

Comment:
We can take `Vec<(IdentityId, bool)>` and input and that will mean that we don't have to do the length check.